pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_status.go

```diff
@@ -0,0 +1,106 @@
+package gatewayapi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_schema "k8s.io/apimachinery/pkg/runtime/schema"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func (r *HTTPRouteReconciler) findReference(gk kube_schema.GroupKind) (kube_client.Object, error) {
+	versions := r.Scheme.VersionsForGroupKind(
+		kube_schema.GroupKind{Group: gk.Group, Kind: gk.Kind},
+	)
+
+	if len(versions) == 0 {
+		return nil, nil
+	}
+
+	gvk := kube_schema.GroupVersionKind{
+		Group:   gk.Group,
+		Kind:    gk.Kind,
+		Version: versions[0].Version,
+	}
+
+	rawObj, err := r.Scheme.New(gvk)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, ok := rawObj.(kube_client.Object)
+	if !ok {
+		return nil, nil
+	}
+
+	return obj, nil
+}
+
+func (r *HTTPRouteReconciler) resolvedRefCondition(
+	ctx context.Context, route kube_client.Object, ref gatewayapi.ParentRef,
+) (kube_meta.Condition, error) {
+	namespace := route.GetNamespace()
+	if ns := ref.Namespace; ns != nil {
+		namespace = string(*ns)
+	}
+
+	resolvedStatus := kube_meta.ConditionTrue
+	resolvedReason := gatewayapi.ConditionRouteResolvedRefs
+
+	obj, err := r.findReference(kube_schema.GroupKind{Group: string(*ref.Group), Kind: string(*ref.Kind)})
+	if err != nil {
+		return kube_meta.Condition{}, err
+	}
+
+	if obj != nil {
+		if err := r.Client.Get(ctx, kube_types.NamespacedName{Namespace: namespace, Name: string(ref.Name)}, obj); err != nil {
+			if apierrors.IsNotFound(err) {
+				resolvedStatus = kube_meta.ConditionFalse
+				resolvedReason = "ObjectNotFound"
+			} else {
+				return kube_meta.Condition{}, err
+			}
+		}
+	} else {
+		resolvedStatus = kube_meta.ConditionFalse
+		resolvedReason = "ObjectTypeUnknownOrInvalid"
+	}
+
+	return kube_meta.Condition{
+		Type: string(gatewayapi.ConditionRouteResolvedRefs), Status: resolvedStatus, Reason: string(resolvedReason),
+		LastTransitionTime: kube_meta.Now(), ObservedGeneration: route.GetGeneration(),
+	}, nil
+}
+
+func (r *HTTPRouteReconciler) updateStatus(ctx context.Context, route *gatewayapi.HTTPRoute) error {
+	var statuses []gatewayapi.RouteParentStatus
+
+	for _, ref := range route.Spec.ParentRefs {
+		resolved, err := r.resolvedRefCondition(ctx, route, ref)
```

Comment:
If the parent of the route wasn't a gateway that we acre controlling, then we should not be updating status (some other controller should do that). It's harmless (but a bit messy) to create the GatewayRoute unconditionally since it would never match any Kuma Gateway.

Reply:
Yes I was depending on the fact that it's harmless to create the `GatewayRoute` but very good point about multiple controllers accessing this.

Reply:
It's actually a little unclear who should be setting `ResolvedRefs` `false` at all, since if we can resolve the ref, only the controller responsible for the `Gateway` should be setting the status. If we can't resolve the ref, either all controllers will set this status or none can. I think we can only set it if it's `true`.

Reply:
Rewritten